lib/terrafying/components/service.rb

```diff
@@ -87,16 +87,16 @@ def create_in(vpc, name, options={})
         depends_on = options[:depends_on] + options[:keypairs].map{ |kp| kp[:resources] }.flatten
 
         iam_statements = options[:iam_policy_statements] + options[:keypairs].map { |kp| kp[:iam_statement] }
-        instance_profile = add! InstanceProfile.create(ident, { statements: iam_statements })
+        @instance_profile = add! InstanceProfile.create(ident, { statements: iam_statements })
 
         tags = options[:tags].merge({ service_name: name })
 
         set = options[:instances].is_a?(Hash) ? DynamicSet : StaticSet
 
-        wants_load_balancer = (set == DynamicSet && @ports.count > 0) || options[:loadbalancer]
+        wants_load_balancer = options.has_key?(:loadbalancer) ? options[:loadbalancer] : (set == DynamicSet && @ports.count > 0)
```

Comment:
`:loadbalancer` should be a boolean (maybe the name should be better), controlling whether we want a load balancer associated with the service. 

Previously, if you had an ASG and want to expose some ports it would force you into having a load balancer. In the context of the Consul cluster, we want an ASG, with exposed ports, but no load balancer. 

`||` is commutative so I'm not sure how switching it around would have any change. Have I missed something?

Reply:
I see where I made the mistake then, I expected the loadbalancer to be the config for a loadbalancer itself not a boolean flagging to create one. In which case, carry on. :)